File: app/schemas/sync.py
from pydantic import BaseModel


class PimSyncResponse(BaseModel):
    products_created: int = 0
    products_updated: int = 0
    products_unchanged: int = 0
    variants_created: int = 0
    variants_updated: int = 0
    variants_unchanged: int = 0
    images_discovered: int = 0
    images_synced: int = 0

File: frontend/lib/types/sync.ts
export interface SyncRun {
  id: number;
  started_at: string;
  finished_at: string | null;
  products_created: number;
  products_updated: number;
  variants_created: number;
  variants_updated: number;
  images_synced: number;
  status: "success" | "failed" | "running";
  error_message: string | null;
}

export interface PimSyncResponse {
  products_created: number;
  products_updated: number;
  products_unchanged: number;
  variants_created: number;
  variants_updated: number;
  variants_unchanged: number;
  images_discovered: number;
  images_synced: number;
  records_processed: number;
  records_created: number;
  records_updated: number;
}
